core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -689,7 +689,15 @@ private[spark] class AppStatusListener(
       if (metricsDelta != null) {
         stage.metrics = LiveEntityHelpers.addMetrics(stage.metrics, metricsDelta)
       }
-      stage.activeTasks -= 1
+      // SPARK-41187: For `SparkListenerTaskEnd` with `Resubmitted` reason, which is raised by
+      // executor lost, it can lead to negative `LiveStage.activeTasks` since there's no
+      // corresponding `SparkListenerTaskStart` event for each of them. The negative activeTasks
+      // will make the stage always remains in the live stage list as it can never meet the
+      // condition activeTasks == 0. This in turn causes the dead executor to never be retained
+      // if that live stage's submissionTime is less than the dead executor's removeTime.
+      if (event.reason != Resubmitted) {
```

Comment:
Can we extend xxxDelta above to include`Resubmitted` reason (e.g., activeTaskDelta) so we don't have to check `event.reason` every time below?